File: app/src/main/java/com/nicholasworkshop/tinklabstest/activity/MainFragmentPagerAdapter.java
package com.nicholasworkshop.tinklabstest.activity;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.nicholasworkshop.tinklabstest.fragment.guide.GuideFragment;
import com.nicholasworkshop.tinklabstest.fragment.guide.GuideModule;

class MainFragmentPagerAdapter extends FragmentPagerAdapter {

    MainFragmentPagerAdapter(FragmentManager fragmentManager) {
        super(fragmentManager);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return GuideFragment.newInstance(GuideModule.TYPE_CITY);
            case 1:
                return GuideFragment.newInstance(GuideModule.TYPE_SHOP);
            case 2:
                return GuideFragment.newInstance(GuideModule.TYPE_EAT);
        }
        return null;
    }

    @Override
    public int getCount() {
        return 3;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "CITY GUIDE";
            case 1:
                return "SHOP";
            case 2:
                return "EAT";
        }
        return null;
    }
}

File: app/src/main/kotlin/com/nicholasworkshop/tinklabstest/activity/MainFragmentPagerAdapter.kt
package com.nicholasworkshop.tinklabstest.activity

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.nicholasworkshop.tinklabstest.R
import com.nicholasworkshop.tinklabstest.fragment.guide.*

internal class MainFragmentPagerAdapter(
        private val context: Context,
        fragmentManager: FragmentManager)
    : FragmentPagerAdapter(fragmentManager) {

    override fun getItem(position: Int): Fragment? {
        when (position) {
            0 -> return GuideFragment.newInstance(GuideType.CITY)
            1 -> return GuideFragment.newInstance(GuideType.SHOP)
            2 -> return GuideFragment.newInstance(GuideType.EAT)
        }
        return null
    }

    override fun getCount(): Int {
        return 3
    }

    override fun getPageTitle(position: Int): CharSequence? {
        when (position) {
            0 -> return context.getString(R.string.main_tab_city)
            1 -> return context.getString(R.string.main_tab_shop)
            2 -> return context.getString(R.string.main_tab_eat)
        }
        return null
    }
}
